mlflow/projects/databricks.py

```diff
@@ -120,20 +120,20 @@ def _upload_project_to_dbfs(self, project_dir, experiment_id):
                             a directory containing an MLproject file).
         """
         temp_tarfile_dir = tempfile.mkdtemp()
-        temp_tar_filename = os.path.join(temp_tarfile_dir, "project.tar.gz")
+        temp_tar_filename = posixpath.join(temp_tarfile_dir, "project.tar.gz")
 
         def custom_filter(x):
-            return None if os.path.basename(x.name) == "mlruns" else x
+            return None if posixpath.basename(x.name) == "mlruns" else x
```

Comment:
```suggestion
            return None if os.path.basename(x.name) == "mlruns" else x
```

As above, we should use the OS's path logic for local files